code/modules/mining/lavaland/loot/legion_loot.dm

```diff
@@ -59,6 +91,13 @@
 			animate(user, color = old_color, transform = old_transform, time = 1 SECONDS)
 
 /obj/item/storm_staff/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	// This early return stops the staff from shooting lightning at someone when being used as a lighter.
+	if(iscarbon(target))
+		var/mob/living/carbon/cig_haver = target
+		var/mask_item = cig_haver.get_item_by_slot(SLOT_HUD_WEAR_MASK)		
+		if(istype(mask_item, /obj/item/clothing/mask/cigarette) && user.zone_selected == "mouth" && user.a_intent == INTENT_HELP)
+			return
```

Comment:
Can we not use the cig check here?

Reply:
This causes the act to fire twice on click, so it will always complain that you've already lit the cigarette when you light it.

Reply:
What will cause the second call of that proc then?

Reply:
I forgot I could just `return TRUE` to cancel `afterattack()`. Issue resolved, one check only.